modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx

```diff
@@ -688,6 +704,16 @@ export default function ObjectFieldFormBase({
 							onToggle={(required) => {
 								setValues({required});
 
+								if (
+									setDbObjectFieldRequired &&
+									dbObjectFieldRequired &&
+									editingObjectField &&
+									objectDefinition?.status?.label ===
+										'approved'
+								) {
+									setDbObjectFieldRequired(required);
+								}
```

Comment:
Great @igor-franca, but with my new proposal I had to add the `modelBuilder` prop to this condition, can you test this please?

```tsx
if (
	setDbObjectFieldRequired &&
	dbObjectFieldRequired &&
	editingObjectField &&
	modelBuilder &&
	objectDefinition?.status?.label ===
		'approved'
) {
	setDbObjectFieldRequired(required);
}
```